typescript/cli/src/commands/warp.ts

```diff
@@ -60,6 +63,75 @@ export const warpCommand: CommandModule = {
   handler: () => log('Command required'),
 };
 
+export const apply: CommandModuleWithWriteContext<{
+  config: string;
+  symbol?: string;
+  warp: string;
+}> = {
+  command: 'apply',
+  describe: 'Update Warp Route contracts',
+  builder: {
+    config: warpDeploymentConfigCommandOption,
+    symbol: {
+      ...symbolCommandOption,
+      demandOption: false,
+    },
+    warp: {
+      ...warpCoreConfigCommandOption,
+      demandOption: true,
+    },
+  },
+  handler: async ({ context, config, symbol, warp }) => {
+    let warpCoreConfig: WarpCoreConfig;
+    if (symbol) {
+      warpCoreConfig = await selectRegistryWarpRoute(context.registry, symbol);
+    } else if (warp) {
+      warpCoreConfig = readWarpCoreConfig(warp);
+    } else {
+      logRed(`Please specify either a symbol or warp config`);
+      process.exit(0);
+    }
+    // Convert warpCoreConfig.tokens into a mapping of { [chainName]: Config } to allow O(1) reads
+    const warpCoreReduced = warpCoreConfig.tokens.reduce<
+      Record<string, WarpCoreConfig['tokens'][number]>
+    >((o, config) => {
+      o[config.chainName] = config;
+      return o;
+    }, {});
+
+    const warpDeployConfig = await readWarpRouteDeployConfig(config);
+
+    // Fetch the static Ism factories and attach to WarpDeploy
+    const addresses = await context.registry.getAddresses();
+    const updates = await promiseObjAll(
+      objMap(warpDeployConfig, async (chain, config) => {
+        config.ismFactoryAddresses = addresses[chain] as any;
+        const evmERC20WarpModule = new EvmERC20WarpModule(
+          context.multiProvider,
+          {
+            config,
+            chain,
+            addresses: {
+              deployedTokenRoute: warpCoreReduced[chain as any].addressOrDenom!,
+            },
+          },
+        );
+        return evmERC20WarpModule.update(config);
+      }),
+    );
+
+    // Loop through each chain, and then loop through each update. ugh.
+    // @TODO maybe use concurrency for each chain
```

Comment:
promise per chain should be fine, as long as we catch errors and log them appropriately and display the chains that a deployment was successful or failed on